pageserver/src/tenant/timeline.rs

```diff
@@ -3396,10 +3438,48 @@ impl Timeline {
 
         // deletion will happen later, the layer file manager sets wanted_garbage_collected
 
-        guard.finish_compact_l0(&layer_removal_cs, remove_layers, &insert_layers)?;
+        guard.finish_compact_l0(
+            &layer_removal_cs,
+            remove_layers,
+            &insert_layers,
+            &duplicates,
+        )?;
+
+        if !possibly_overlapping_overwritten.is_empty() {
+            use std::fmt::Write as _;
+            // will do a better version of selection soon after a test run
```

Comment:
Is this still planned?

Reply:
Yes, I need to ask Bojan for help. It should be doable.

Reply:
After looking around and talking to Bojan, if it would help, it would help only a tiny bit. So if this is the way to go, it's just easier to loop over all layers.

Will cleanup the comment perhaps, though thinking alternatives for ensuring the layers are resident.